### ISSUE TITLE
in HTML output, the "self" link on a section title uses the slugifiedName, not the author assigned anchor

### ISSUE THREAD

**Issue by reschke** [ietf-tools/xml2rfc#894]
### Describe the issue

Example from <https://www.rfc-editor.org/rfc/rfc9110.html#uri.references>:

> [4.1. ](https://www.rfc-editor.org/rfc/rfc9110.html#section-4.1)[URI References](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-references)

"URI References" links to "#name-uri-references", not to the auhor-supplied anchor "uri.references".

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by kesara** [comment]
This is done in preptool to avoid overlapping values as part of slugyfy names [1].

The RFC 7991 says:
> "slugifiedName" always has the format "n-ttt", where "ttt" is the
      text of the name after slugification.  For example, this would be
      "n-protocol-overview" for the name "Protocol Overview".  The
      actual conversions done in slugification will be specified at a
      later time.

But in xml2rfc this is implemented as `name-` instead of `n-`.

[1] https://www.rfc-editor.org/rfc/rfc7991.html#appendix-B.2.1

**Comment by kesara** [comment]
I looked at the code history and looks like `xml2rfc` has always used `sluggifiedName` for the `anchor`.
So the correct logic to identify `anchor` should be:
```
if anchor is set:
  use the anchor
else:
  use sluggifiedName
```

**Comment by reschke** [comment]
Right. The easiest way to discover the author supplied link is to click on the section title, and that really needs to go to the proper anchor.
